coderd/prometheusmetrics/aggregator.go

```diff
@@ -64,6 +67,20 @@ type annotatedMetric struct {
 	expiryDate time.Time
 }
 
+func hashKey(req *updateRequest, m *agentproto.Stats_Metric) string {
+	var sbLabels strings.Builder
+	for i, label := range m.GetLabels() {
+		_, _ = sbLabels.WriteString(label.Name)
+		_ = sbLabels.WriteByte('=')
+		_, _ = sbLabels.WriteString(label.Value)
+
+		if i-1 != len(m.GetLabels()) {
```

Comment:
Suggestion: You could trim `,` at the end of the loop to avoid the in-loop conditional.